src/allmydata/scripts/create_node.py

```diff
@@ -19,10 +20,38 @@ def write_tac(basedir, nodetype):
     fileutil.write(os.path.join(basedir, "tahoe-%s.tac" % (nodetype,)), dummy_tac)
 
 
+WHERE_OPTS = [
+    ("location", None, None, "Specify the location to advertise for this node."),
+    ("port", None, None, "Specify the server endpoint to listen on for this node."),
+]
+
+def validate_where_options(options):
+    if options['hostname'] and options['port']:
+        raise usage.UsageError("The --hostname option cannot be used with the --port option.")
+    if options['hostname'] and options['location']:
+        raise usage.UsageError("The --hostname option cannot be used with the --location option.")
+    if not options['hostname'] and (options['location'] and not options['port']):
+        raise usage.UsageError("The --location option must be used with the --port option.")
+    if not options['hostname'] and (options['port'] and not options['location']):
+        raise usage.UsageError("The --port option must be used with the --location option.")
+    if (options['listen'] != "tcp") and options['hostname']:
+        raise usage.UsageError("The listener type must be TCP to use --hostname option.")
+
 class _CreateBaseOptions(BasedirOptions):
     optFlags = [
         ("hide-ip", None, "prohibit any configuration that would reveal the node's IP address"),
         ]
+
+    # This is overridden in order to ensure we get a "Wrong number of
+    # arguments." error when more than one argument is given.
+    def parseArgs(self, basedir=None):
+        BasedirOptions.parseArgs(self, basedir)
```

Comment:
(The reason why the method needs to be overridden in the Options subclass is that twisted.python.usage.Options introspects the method in that class to check the number of arguments, and that doesn't work if it's inherited.)